File: Algorithms/189. Rotate Array/javascript/solution.extra_array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function (nums, k) {
    const n = nums.length;
    const a = new Array(n).fill(0);
    for (let i = 0; i < n; i++) {
        a[(i + k) % n] = nums[i];
    }
    for (let i = 0; i < n; i++) {
        nums[i] = a[i];
    }
};

File: Algorithms/189. Rotate Array/golang/solution.extra_array.go
func rotate(nums []int, k int) {
	var n int = len(nums)
	var a []int = make([]int, n)
	for i := 0; i < n; i++ {
		a[(i+k)%n] = nums[i]
	}
	for i := 0; i < n; i++ {
		nums[i] = a[i]
	}
}
